src/make_round_config.R

```diff
@@ -1 +1,162 @@
+  if (isFALSE(weekdays(as.Date(this_round_date)) == "Wednesday")) {
+    stop("The latest clade_file does not have a Wednesday date: ", this_round_clade_file)
+  }
+
+  model_tasks <- hubAdmin::create_model_tasks(
+    hubAdmin::create_model_task(
+      task_ids = hubAdmin::create_task_ids(
+        hubAdmin::create_task_id("nowcast_date",
+          required = list(this_round_date),
+          optional = NULL
+        ),
+        hubAdmin::create_task_id("target_date",
+          required = NULL,
+          ## target date is nowcast_date and the three prior weeks
+          optional = as.character(as.Date(this_round_date) - c(0, 7, 14, 21))
+        ),
+        hubAdmin::create_task_id("location",
+          required = NULL,
+          optional = c(
+            "AL", "AK", "AZ", "AR", "CA", "CO",
+            "CT", "DE", "DC", "FL", "GA", "HI",
+            "ID", "IL", "IN", "IA", "KS", "KY",
+            "LA", "ME", "MD", "MA", "MI", "MN",
+            "MS", "MO", "MT", "NE", "NV", "NH",
+            "NJ", "NM", "NY", "NC", "ND", "OH",
+            "OK", "OR", "PA", "RI", "SC", "SD",
+            "TN", "TX", "UT", "VT", "VA", "WA",
+            "WV", "WI", "WY", "PR"
+          )
+        ),
+        hubAdmin::create_task_id("variant",
+          required = this_round_clade_list,
+          optional = NULL
+        )
+      ),
+      output_type = hubAdmin::create_output_type(
+        hubAdmin::create_output_type_sample(
+          is_required = FALSE,
+          output_type_id_type = "character",
+          max_length = 15L,
+          min_samples_per_task = 1L,
+          max_samples_per_task = 500L,
+          value_type = "double",
+          value_minimum = 0L,
+          value_maximum = 1L,
+        )
+      ),
+      target_metadata = hubAdmin::create_target_metadata(
+        hubAdmin::create_target_metadata_item(
+          target_id = "variant prop",
+          target_name = "Weekly nowcasted variant proportions",
+          target_units = "proportion",
+          target_keys = NULL,
+          target_type = "compositional",
+          is_step_ahead = TRUE,
+          time_unit = "week"
+        )
+      )
+    )
+  )
+
+  round <- hubAdmin::create_round(
+    "nowcast_date",
+    round_id_from_variable = TRUE,
+    model_tasks = model_tasks,
+    submissions_due = list(
+      relative_to = "nowcast_date",
+      start = -7L,
+      end = 1L
+    )
+  )
+  return(round)
+}
+
+write_and_validate_task_config <- function(task_config, round_id) {
+  hub_dir <- dirname(getwd())
+  hubAdmin::write_config(task_config, hub_path = hub_dir, overwrite = TRUE, silent = TRUE)
+  valid_task_config <- hubAdmin::validate_config(
+    hub_path = hub_dir,
+    config = c("tasks"),
+    schema_version = "from_config",
+    branch = "main"
+  )
+  if (isFALSE(valid_task_config)) {
+    print(valid_task_config)
+    stop("Generated task config is invalid for round: ", round_id)
+  }
+}
+
+coerce_to_round <- function(existing_round, new_round) {
```

Comment:
See this `hubverse` thread for more background on the `coerce_to_round` function: https://reichlab.slack.com/archives/C02QMRCGA4F/p1726594089303389